hack/build/ci/prepare-build-variables.sh

```diff
@@ -3,6 +3,8 @@
 create_docker_image_tag() {
   if [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]]; then
     echo "snapshot-${GITHUB_HEAD_REF//[^a-zA-Z0-9_-]/-}"; return
+  elif [[ "${GITHUB_EVENT_NAME}" == "schedule" ]]; then # nightly builds
+    echo "nightly-$(date --iso-8601)-${GITHUB_SHA}"; return
```

Comment:
how will you "find" this image/helm chart later, when you want to automatically use them in the e2e tests?

do we need the `${GITHUB_SHA}` in that? couldn't it be just a label on the image instead?

Reply:
technically, the sha **should** remain the same, between the nightly build running and the e2e tests running afterwards, but what if there is a sneaky `renovate` PR the flies in, changing the sha, so the e2e tests won't find what it is looking for 👀

Reply:
Valid concerns; I've tried to address them in b8aef6284